src/server/job_func.c

```diff
@@ -958,18 +1012,8 @@ job_purge(job *pjob)
 #endif	/* PBS_MOM */
 
 #ifdef PBS_MOM
-	(void)strcpy(namebuf, path_jobs);	/* delete job file */
-	if (*pjob->ji_qs.ji_fileprefix != '\0')
-		(void)strcat(namebuf, pjob->ji_qs.ji_fileprefix);
-	else
-		(void)strcat(namebuf, pjob->ji_qs.ji_jobid);
-	(void)strcat(namebuf, JOB_FILE_SUFFIX);
-	if (unlink(namebuf) < 0) {
-		if (errno != ENOENT) {
-			log_joberr(errno, __func__, msg_err_purgejob,
-				pjob->ji_qs.ji_jobid);
-		}
-	}
+	/* delete job file */
+	del_job_related_file(pjob, JOB_FILE_SUFFIX);
```

Comment:
I think we have to do delete_cred and delete of `.JB` files (see line no. 1018) in mock run also...
Maybe you can create one function like `del_job_creds()` which will call `delete_cred()` as well as delete `.JB` files and call it here as well as in mock_run_job_purge()

Also, I know this won't affect normal mom run, but just to ensure that we are not breaking anything... can you please run a regression with these changes?

Reply:
I'm already deleting .JB, del_job_related_file() will do it when i pass JOB_FILE_SUFFIX right?  I've added a call to delete_cred(), let me know what you think.

I'll do a full regression, thanks.

Reply:
Sorry I mean, JOB_CRED_SUFFIX not JOB_FILE_SUFFIX.
https://github.com/PBSPro/pbspro/blob/b9ed6cf4bd108c3ef3d4da87e061e82514ee4a44/src/server/job_func.c#L1062-L1073

Reply:
but that code is not executed for mom, it's not under #ifdef PBS_MOM, right?

Reply:
I think that code is for both server as well as mom because AFAIK both will have cred file for job